pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test: TestLogic_trigram_builtins failed

pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test.TestLogic_trigram_builtins [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11039122?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11039122?buildTab=artifacts#/) on master @ [06fb4c1b02fc90f8ef17ec1342bb6ce600278925](https://github.com/cockroachdb/cockroach/commits/06fb4c1b02fc90f8ef17ec1342bb6ce600278925):


```
        "COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING=false" to disable all random test variables altogether.
*
* WARNING: test tenant requested by configuration, but code organization prevents start!
* OSS binaries do not include enterprise features
*
    logic.go:2925: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7695/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins:45: SELECT similarity(a, b) FROM (VALUES
            ('', ''),
            ('foo', ''),
            ('', 'foo'),
            ('foo', NULL),
            (NULL, 'foo'),
            (NULL, NULL),
            ('foo', 'bar'),
            ('foo', 'far'),
            ('foo', 'for'),
            ('foo', 'foo')
          ) tbl(a, b)
        expected:
            0
            0
            0
            NULL
            NULL
            NULL
            0
            0.142857142857143
            0.333333333333333
            1
            
        but found (query options: "nosort") :
            0
            0
            0
            NULL
            NULL
            NULL
            0
            0.14285714285714285
            0.3333333333333333
            1
            
[17:03:32] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7695/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins with config local-mixed-22.2-23.1: 6 tests, 1 failures
    logic.go:3983: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7695/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins:70: error while processing
    logic.go:3983: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7695/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins:70: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5e67fe52d711043100a1ff1e2183dfca/logTestLogic_trigram_builtins2883590916
--- FAIL: TestLogic_trigram_builtins (0.65s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_trigram_builtins.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30056

pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test.TestLogic_trigram_builtins [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11039474?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11039474?buildTab=artifacts#/) on master @ [c917a6bd896730c86867e2ed6399725ec10e0ac2](https://github.com/cockroachdb/cockroach/commits/c917a6bd896730c86867e2ed6399725ec10e0ac2):


```
        "COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING=false" to disable all random test variables altogether.
*
* WARNING: test tenant requested by configuration, but code organization prevents start!
* OSS binaries do not include enterprise features
*
    logic.go:2925: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/23/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins:45: SELECT similarity(a, b) FROM (VALUES
            ('', ''),
            ('foo', ''),
            ('', 'foo'),
            ('foo', NULL),
            (NULL, 'foo'),
            (NULL, NULL),
            ('foo', 'bar'),
            ('foo', 'far'),
            ('foo', 'for'),
            ('foo', 'foo')
          ) tbl(a, b)
        expected:
            0
            0
            0
            NULL
            NULL
            NULL
            0
            0.142857142857143
            0.333333333333333
            1
            
        but found (query options: "nosort") :
            0
            0
            0
            NULL
            NULL
            NULL
            0
            0.14285714285714285
            0.3333333333333333
            1
            
[17:36:26] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/23/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins with config local-mixed-22.2-23.1: 6 tests, 1 failures
    logic.go:3983: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/23/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins:70: error while processing
    logic.go:3983: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/23/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins:70: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5e67fe52d711043100a1ff1e2183dfca/logTestLogic_trigram_builtins3216878402
--- FAIL: TestLogic_trigram_builtins (0.72s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_trigram_builtins.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test.TestLogic_trigram_builtins [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11040491?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11040491?buildTab=artifacts#/) on master @ [c917a6bd896730c86867e2ed6399725ec10e0ac2](https://github.com/cockroachdb/cockroach/commits/c917a6bd896730c86867e2ed6399725ec10e0ac2):


```
=== RUN   TestLogic_trigram_builtins
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/5e67fe52d711043100a1ff1e2183dfca/logTestLogic_trigram_builtins1416715066
    test_log_scope.go:81: use -show-logs to present logs inline
[18:52:24] setting distsql_workmem='78027B';
    logic.go:2925: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/6168/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins:45: SELECT similarity(a, b) FROM (VALUES
            ('', ''),
            ('foo', ''),
            ('', 'foo'),
            ('foo', NULL),
            (NULL, 'foo'),
            (NULL, NULL),
            ('foo', 'bar'),
            ('foo', 'far'),
            ('foo', 'for'),
            ('foo', 'foo')
          ) tbl(a, b)
        expected:
            0
            0
            0
            NULL
            NULL
            NULL
            0
            0.142857142857143
            0.333333333333333
            1
            
        but found (query options: "nosort") :
            0
            0
            0
            NULL
            NULL
            NULL
            0
            0.14285714285714285
            0.3333333333333333
            1
            
[18:52:24] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/6168/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins with config local-mixed-22.2-23.1: 6 tests, 1 failures
    logic.go:3983: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/6168/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins:70: error while processing
    logic.go:3983: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/6168/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins:70: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
--- FAIL: TestLogic_trigram_builtins (0.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_trigram_builtins.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test.TestLogic_trigram_builtins [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11040507?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11040507?buildTab=artifacts#/) on master @ [4618dbf07f1fe42e26e16c95b1b495119ad5d267](https://github.com/cockroachdb/cockroach/commits/4618dbf07f1fe42e26e16c95b1b495119ad5d267):


```
        "COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING=false" to disable all random test variables altogether.
*
* WARNING: test tenant requested by configuration, but code organization prevents start!
* OSS binaries do not include enterprise features
*
    logic.go:2925: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9028/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins:45: SELECT similarity(a, b) FROM (VALUES
            ('', ''),
            ('foo', ''),
            ('', 'foo'),
            ('foo', NULL),
            (NULL, 'foo'),
            (NULL, NULL),
            ('foo', 'bar'),
            ('foo', 'far'),
            ('foo', 'for'),
            ('foo', 'foo')
          ) tbl(a, b)
        expected:
            0
            0
            0
            NULL
            NULL
            NULL
            0
            0.142857142857143
            0.333333333333333
            1
            
        but found (query options: "nosort") :
            0
            0
            0
            NULL
            NULL
            NULL
            0
            0.14285714285714285
            0.3333333333333333
            1
            
[18:54:06] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9028/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins with config local-mixed-22.2-23.1: 6 tests, 1 failures
    logic.go:3983: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9028/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins:70: error while processing
    logic.go:3983: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9028/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins:70: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5e67fe52d711043100a1ff1e2183dfca/logTestLogic_trigram_builtins64484466
--- FAIL: TestLogic_trigram_builtins (0.58s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_trigram_builtins.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test.TestLogic_trigram_builtins [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11041414?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11041414?buildTab=artifacts#/) on master @ [acc75f4605294f2c872ceb63b12f2c9605711d3b](https://github.com/cockroachdb/cockroach/commits/acc75f4605294f2c872ceb63b12f2c9605711d3b):


```
        "COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING=false" to disable all random test variables altogether.
*
* WARNING: test tenant requested by configuration, but code organization prevents start!
* OSS binaries do not include enterprise features
*
    logic.go:2925: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11305/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins:45: SELECT similarity(a, b) FROM (VALUES
            ('', ''),
            ('foo', ''),
            ('', 'foo'),
            ('foo', NULL),
            (NULL, 'foo'),
            (NULL, NULL),
            ('foo', 'bar'),
            ('foo', 'far'),
            ('foo', 'for'),
            ('foo', 'foo')
          ) tbl(a, b)
        expected:
            0
            0
            0
            NULL
            NULL
            NULL
            0
            0.142857142857143
            0.333333333333333
            1
            
        but found (query options: "nosort") :
            0
            0
            0
            NULL
            NULL
            NULL
            0
            0.14285714285714285
            0.3333333333333333
            1
            
[19:51:42] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11305/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins with config local-mixed-22.2-23.1: 6 tests, 1 failures
    logic.go:3983: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11305/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins:70: error while processing
    logic.go:3983: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11305/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/trigram_builtins:70: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5e67fe52d711043100a1ff1e2183dfca/logTestLogic_trigram_builtins1090070100
--- FAIL: TestLogic_trigram_builtins (0.58s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_trigram_builtins.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/testutils/floatcmp/floatcmp.go
+++ b/pkg/testutils/floatcmp/floatcmp.go
@@ -121,9 +121,23 @@ func FloatsMatch(expectedString, actualString string) (bool, error) {
 	actual = math.Abs(actual)
 	// Check that 15 significant digits match. We do so by normalizing the
 	// numbers and then checking one digit at a time.
+	//
+	// normalize converts f to base * 10**power representation where base is in
+	// [1.0, 10.0) range.
+	normalize := func(f float64) (base float64, power int) {
+		for f >= 10 {
+			f = f / 10
+			power++
+		}
+		for f < 1 {
+			f *= 10
+			power--
+		}
+		return f, power
+	}
 	var expPower, actPower int
-	expected, expPower = math.Frexp(expected)
-	actual, actPower = math.Frexp(actual)
+	expected, expPower = normalize(expected)
+	actual, actPower = normalize(actual)
 	if expPower != actPower {
 		return false, nil
 	}

--- a/pkg/testutils/floatcmp/floatcmp_test.go
+++ b/pkg/testutils/floatcmp/floatcmp_test.go
@@ -178,6 +178,7 @@ func TestFloatsMatch(t *testing.T) {
 		{f1: "0.1234567890123456", f2: "0.1234567890123457", match: true},
 		{f1: "-0.1234567890123456", f2: "0.1234567890123456", match: false},
 		{f1: "-0.1234567890123456", f2: "-0.1234567890123455", match: true},
+		{f1: "0.142857142857143", f2: "0.14285714285714285", match: true},
 	} {
 		match, err := FloatsMatch(tc.f1, tc.f2)
 		if err != nil {

